extension_service/datastore/providers/postgres.py

```diff
@@ -198,6 +198,52 @@ async def get_airport(self, id: int) -> Optional[models.Airport]:
         result = models.Airport.model_validate(dict(result))
         return result
 
+    async def get_airport_by_iata(self, iata: str) -> Optional[models.Airport]:
+        result = await self.__pool.fetchrow(
+            """
+              SELECT * FROM airports WHERE iata ILIKE $1
+            """,
+            iata,
+        )
+
+        if result is None:
+            return None
+
+        result = models.Airport.model_validate(dict(result))
+        return result
+
+    async def search_airports(
+        self,
+        country: Optional[str] = None,
+        city: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> Optional[list[models.Airport]]:
+        if country is None:
+            country = "%"
+        if city is None:
+            city = "%"
+        name = "%" if name is None else "%" + name + "%"
+
+        results = await self.__pool.fetch(
+            """
+            SELECT * FROM (
+                SELECT * FROM (
+                    SELECT * FROM airports
+                    WHERE country ILIKE $1
+                ) AS filtered_country
+                WHERE city ILIKE $2
+            ) AS filtered_city
+            WHERE name ILIKE $3
```

Comment:
This seemed like too much nesting, so I asked Bard and it recommend this:

```
SELECT * FROM users
WHERE ($1 IS NULL OR country ILIKE $1)
AND ($2 NULL OR city ILIKE $2)
AND ($3 IS NULL OR name LIKE '%$3%')
```

(Which seems pretty good to me if it works)